src/sinks/aws_kinesis_firehose.rs

```diff
@@ -0,0 +1,374 @@
+
+        assert_eq!(&event.data[..], message.as_bytes());
+    }
+
+    #[test]
+    fn firehose_encode_event_json() {
+        let message = "hello world".to_string();
+        let mut event = Event::from(message.clone());
+        event
+            .as_mut_log()
+            .insert_explicit("key".into(), "value".into());
+        let event = encode_event(event, &Encoding::Json).unwrap();
+
+        let map: HashMap<String, String> = serde_json::from_slice(&event.data[..]).unwrap();
+
+        assert_eq!(map[&event::MESSAGE.to_string()], message);
+        assert_eq!(map["key"], "value".to_string());
+    }
+}
+
+#[cfg(feature = "firehose-integration-tests")]
+#[cfg(test)]
+mod integration_tests {
+    use super::*;
+    use crate::{
+        region::RegionOrEndpoint,
+        runtime,
+        sinks::elasticsearch::{ElasticSearchCommon, ElasticSearchConfig, Provider},
+        test_util::{random_events_with_stream, random_string},
+        topology::config::SinkContext,
+    };
+    use futures::Sink;
+    use rusoto_core::Region;
+    use rusoto_firehose::{
+        CreateDeliveryStreamInput, ElasticsearchDestinationConfiguration, KinesisFirehose,
+        KinesisFirehoseClient,
+    };
+    use serde_json::{json, Value};
+    use std::{thread, time::Duration};
+
+    #[test]
+    fn firehose_put_records() {
+        let stream = gen_stream();
+
+        let region = Region::Custom {
+            name: "localstack".into(),
+            endpoint: "http://localhost:4573".into(),
+        };
+
+        ensure_stream(region.clone(), stream.clone());
+
+        let config = KinesisFirehoseSinkConfig {
+            stream_name: stream.clone(),
+            region: RegionOrEndpoint::with_endpoint("http://localhost:4573".into()),
+            encoding: Encoding::Json, // required for ES destination w/ localstack
+            batch: BatchConfig {
+                batch_size: Some(2),
+                batch_timeout: None,
+            },
+            request: TowerRequestConfig {
+                request_timeout_secs: Some(10),
+                request_retry_attempts: Some(0),
+                ..Default::default()
+            },
+        };
+
+        let mut rt = runtime::Runtime::new().unwrap();
+        let cx = SinkContext::new_test(rt.executor());
+
+        let sink = KinesisFirehoseService::new(config, cx).unwrap();
+
+        let (input, events) = random_events_with_stream(100, 100);
+
+        let pump = sink.send_all(events);
+        let _ = rt.block_on(pump).unwrap();
+
+        thread::sleep(Duration::from_secs(1));
+
+        let config = ElasticSearchConfig {
+            provider: Some(Provider::Aws),
+            region: RegionOrEndpoint::with_endpoint("http://localhost:4571".into()),
+            index: Some(stream.clone()),
+            ..Default::default()
+        };
+        let common = ElasticSearchCommon::parse_config(&config).expect("Config error");
+
+        let client = reqwest::Client::builder()
```

Comment:
I kinda wanna get rid of reqwest and just use the elastic libraries serde feature. But we can do that in the future.